packages/sdks/snippets/angular-16/src/app/live-preview/live-preview.component.ts

```diff
@@ -0,0 +1,61 @@
+import { CommonModule } from '@angular/common';
+import type { OnDestroy, OnInit } from '@angular/core';
+import { Component, inject } from '@angular/core';
+import { Router } from '@angular/router';
+import type { BuilderContent } from '@builder.io/sdk-angular';
+import { fetchOneEntry, subscribeToEditor } from '@builder.io/sdk-angular';
+
+@Component({
+  standalone: true,
+  selector: 'app-live-preview',
+  imports: [CommonModule],
+  template: `
+    <div *ngIf="loading">Loading Data...</div>
+
+    <div *ngIf="!loading && content" class="blog-data-preview">
+      <div>Blog Title: {{ content.data?.title }}</div>
+      <div>Authored by: {{ content.data?.['author'] }}</div>
+      <div>Handle: {{ content.data?.['handle'] }}</div>
+    </div>
+  `,
+})
+export class LivePreviewComponent implements OnInit, OnDestroy {
+  content: BuilderContent | undefined = undefined;
+  loading = false;
+
+  private unsubscribeFn?: () => void;
+  private router = inject(Router);
+
+  getFormattedDate(dateString: string | null): string {
+    return dateString ? new Date(dateString).toDateString() : '';
+  }
+
+  ngOnInit(): void {
+    this.loading = true;
+    fetchOneEntry({
+      model: 'blog-data',
+      apiKey: 'ee9f13b4981e489a9a1209887695ef2b',
+      userAttributes: { urlPath: this.router.url },
+    })
+      .then((data) => {
+        this.content = data || undefined;
```

Comment:
you shouldn't need the `|| undefined` for this

```suggestion
        this.content = data
```

Reply:
Sami, I think `fetchOneEntry` returns a resolved promise of type `Promise<BuilderContent | null>`, and if we don't declare it as `content: BuilderContent | null = null,` it gives us an error. So, I'm thinking of doing the following:

> 
```
   content: BuilderContent | null = null;
  loading = true;

  private unsubscribeFn?: () => void;
  private router = inject(Router);

  ngOnInit(): void {
    fetchOneEntry({
      model: 'blog-data',
      apiKey: 'ee9f13b4981e489a9a1209887695ef2b',
      userAttributes: { urlPath: this.router.url },
    })
      .then((data) => {
        this.content = data;
      })
      .finally(() => {
        this.loading = false;
      });
```